lib/vanagon/component/source/http.rb

```diff
@@ -62,6 +63,16 @@ def initialize(url, sum:, workdir:, sum_type:, **options)
           @sum = sum
           @workdir = workdir
           @sum_type = sum_type
+
+          if Vanagon::Component::Source::Http.valid_url?(@sum)
+            sum_file = download(@sum)
+            File.open(File.join(@workdir, sum_file)) do |file|
+              # the sha1 files generated during archive creation  are formatted
+              # "<sha1sum> <filename>". This will also work for sources that
+              # only contain the checksum.
+              @sum = file.read.split.first
```

Comment:
What happens if `@sum` is set to nil?

Reply:
earlier up in the initialize method we fail unless sum, and nil evaluates to falsey, so it won't get to this

Reply:
if the fetched file is empty, then read will return `""`, split will return `[]` and first will return `nil`.

Reply:
ah, good call. will rework this bit